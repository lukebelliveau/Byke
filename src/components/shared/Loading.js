// @flow
import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { connect } from 'react-redux';

const mapStateToProps = state => ({
  isLoading: state.isLoading,
});

const Loading = ({ isLoading }) =>
  isLoading
    ? <View style={styles.loadingContainer}>
        <View style={styles.loadingBox}>
          <Text style={styles.loadingText}>Loading...</Text>
        </View>
      </View>
    : null;

const styles = StyleSheet.create({
  loadingBox: {
    position: 'absolute',
    bottom: 20,
    elevation: 5,
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'white',
    borderRadius: 20,
    width: '50%',
    padding: 10,
  },
  loadingText: {
    fontSize: 30,
  },
  loadingContainer: {
    position: 'absolute',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    width: '100%',
    height: '100%',
  },
});

export default connect(mapStateToProps)(Loading);
