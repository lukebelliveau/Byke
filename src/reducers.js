const initialState = {
  loading: false,
}
const reducer = (state = initialState, action = {}) => {
  return null;
};

export default reducer;
