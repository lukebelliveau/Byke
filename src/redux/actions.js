export const types = {
  LOADING_FINISHED: 'LOADING_FINISHED',
  LOADING_STARTED: 'LOADING_STARTED',
  STATIONS_FETCHED: 'STATIONS_FETCHED',
};

const loadingStarted = () => ({
  type: types.LOADING_STARTED,
});

const loadingFinished = () => ({
  type: types.LOADING_FINISHED,
});

const stationsFetched = stations => ({
  type: types.STATIONS_FETCHED,
  payload: stations,
});

export default {
  loadingStarted,
  loadingFinished,
  stationsFetched,
};
