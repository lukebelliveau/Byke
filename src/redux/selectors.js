// @flow
import { State } from '../Types';
const getCoordinates = (state: State) => ({
  latitude: state.region.latitude,
  longitude: state.region.longitude,
});

export default {
  getCoordinates,
};
