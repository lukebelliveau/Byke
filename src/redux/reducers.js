import { types } from './actions';

const initialState = {
  isLoading: false,
};
const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.LOADING_STARTED:
      return { isLoading: true };
    case types.LOADING_FINISHED:
      return { isLoading: false };
    default:
      return state;
  }
};

export default reducer;
