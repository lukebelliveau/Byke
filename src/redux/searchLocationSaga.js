// @flow
import { call, put, takeEvery, takeLatest, select } from 'redux-saga/effects';
import api from '../api';
import actions, { types } from './actions';
import selectors from './selectors';

function* fetchLocations(action) {
  const coordinates = yield select(selectors.getCoordinates);
  try {
    const locations = yield call(
      api.searchPlaces,
      action.payload,
      coordinates.latitude,
      coordinates.longitude
    );
    yield put(actions.locationsFetched(locations.results));
  } catch (e) {
    yield put({ type: 'USER_FETCH_FAILED', message: e.message });
  }
}

function* searchLocationSaga(): Generator<*, *, *> {
  yield takeLatest('SEARCH_LOCATIONS', fetchLocations);
}

export default searchLocationSaga;
