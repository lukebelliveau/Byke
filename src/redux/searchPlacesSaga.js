// @flow
import { call, put, takeEvery, takeLatest, select } from 'redux-saga/effects';
import api from '../api';
import actions, { types } from './actions';
import selectors from './selectors';

function* fetchPlaces(action) {
  const coordinates = yield select(selectors.getCoordinates);
  try {
    const places = yield call(
      api.searchPlaces,
      action.payload,
      coordinates.latitude,
      coordinates.longitude
    );
    yield put(actions.placesFetched(places.results));
  } catch (e) {
    yield put({ type: 'USER_FETCH_FAILED', message: e.message });
  }
}

function* searchPlacesSaga(): Generator<*, *, *> {
  yield takeLatest(types.SEARCH_PLACES, fetchPlaces);
}

export default searchPlacesSaga;
