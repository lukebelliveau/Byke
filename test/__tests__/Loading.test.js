import React from 'react';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import Loading from '../../src/Loading';
import reducers from '../../src/redux/reducers';
import actions from '../../src/redux/actions';

it('displays loading message when loading', () => {
  const store = createStore(reducers);
  store.dispatch(actions.loadingStarted());
  const byke = renderer.create(<Provider store={store}><Loading /></Provider>);

  expect(byke).toMatchSnapshot();
});

it('returns nothing when nothing is loading', () => {
  const store = createStore(reducers);
  store.dispatch(actions.loadingFinished());
  const byke = renderer.create(<Provider store={store}><Loading /></Provider>);

  expect(byke).toMatchSnapshot();
});
