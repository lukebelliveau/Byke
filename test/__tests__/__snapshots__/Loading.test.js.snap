// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays loading message when loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "height": "100%",
      "justifyContent": "center",
      "position": "absolute",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 20,
        "bottom": 20,
        "display": "flex",
        "elevation": 5,
        "justifyContent": "center",
        "padding": 10,
        "position": "absolute",
        "width": "50%",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      Loading...
    </Text>
  </View>
</View>
`;

exports[`returns nothing when nothing is loading 1`] = `null`;
